core/src/main/java/io/kestra/core/models/flows/Input.java

```diff
@@ -35,6 +52,8 @@ public class Input {
 
     String defaults;
 
+    public abstract boolean isValid(String input);
```

Comment:
```suggestion
  public abstract void validate(Object input) throws ConstraintViolationException {
```
- To allow a better error message on the ui
- The type can be object I think

Reply:
Will do.
The type is always a string, the cast / parse to the real type of the object is done later. To change this I'll need more changes.

Reply:
I would prefer a proper type, if not, the cast will need to be double done for min or max for example, don't make sense

Reply:
Done